wisp-lease/src/main/kotlin/wisp/lease/pool/PoolLease.kt

```diff
@@ -0,0 +1,53 @@
+package wisp.lease.pool
+
+import wisp.lease.Lease
+
+class PoolLease(
+  private val delegateLease: Lease,
+  private val poolLeaseManager: PoolLeaseManager,
+) : Lease {
+  override val name: String
+    get() = delegateLease.name
```

Comment:
Might be a good use case for delegation:

```suggestion
) : Lease by delegateLease {
```